privacyidea/lib/user.py

```diff
@@ -597,7 +597,22 @@ def get_user_list(param=None, user=None):
     searchDict = {"username": "*"}
     param = param or {}
 
+    # we have to recreate a new searchdict without the realm key
+    # as delete does not work
+    for key in param:
+        lval = param[key]
+        if key == "realm":
+            continue
+        if key == "resolver":
+            continue
+        if key == "user" or key == "username":
+            # If "user" or "username" in param we skip to handle the user afterwards
+            continue
```

Comment:
Sorry for nitpicking and probably my original code was even worse.
But should be do:

~~~~python
for key in param:
    lval = param[key]
    if key in ["realm", "resolver", "user", "username"]:
        continue
    searchDict[key] = lval
    log.debug(...)
~~~~

I was not aware of this earlier.